src/PointEval/point_values.jl

```diff
@@ -59,36 +59,33 @@ function_symmetric_gradient(pv::PointValues, u::AbstractVector, args...) =
 function reinit!(pv::PointValues, x::AbstractVector{<:Vec{D}}, ξ::Vec{D}) where {D}
     qp = 1 # PointValues only have a single qp
     # TODO: Does M need to be updated too?
-    for i in 1:getnbasefunctions(pv.cv.ip)
-        pv.cv.dNdξ[i, qp], pv.cv.N[i, qp] = shape_gradient_and_value(pv.cv.ip, ξ, i)
-    end
+    Nqp = @view pv.cv.N[:, qp]
+    dNdξqp = @view pv.cv.dNdξ[:, qp]
+    shape_gradients_and_values!(dNdξqp, Nqp, pv.cv.ip, ξ)
     reinit!(pv.cv, x)
     return nothing
 end
 
 # Optimized version of PointScalarValues which avoids i) recomputation of dNdξ and
 # ii) recomputation of dNdx. Only allows function evaluation (no gradients) which is
 # what is used in evaluate_at_points.
-struct PointValuesInternal{IP, N_t} <: AbstractValues
-    N::Vector{N_t}
+struct PointValuesInternal{IP, N_t, VT <: AbstractVector{N_t}} <: AbstractValues
+    N::VT
     ip::IP
 end
 
 function PointValuesInternal(ξ::Vec{dim, T}, ip::IP) where {dim, T, shape <: AbstractRefShape{dim}, IP <: Interpolation{shape}}
     n_func_basefuncs = getnbasefunctions(ip)
-    N = [shape_value(ip, ξ, i) for i in 1:n_func_basefuncs]
-    return PointValuesInternal{IP, eltype(N)}(N, ip)
+    N = MVector(ntuple(i->shape_value(ip, ξ, i), n_func_basefuncs))
```

Comment:
Not sure. I am just trying to get stuff ready for GPU mode. :)

Reply:
OK, would it make sense to leave that for a separate issue then?
Instead of hard-coding a different vector type? 

My impression is that other efforts (at least those I'm involved in) are more about allowing GPU-compatible arrays instead of enforcing them?